### PR TITLE
streamingccl,db-console: add Cross-Cluster Replication metrics page

### LINKED ISSUE #95995: c2c: add a cross cluster replication dashboard to the DB console
We would like to have a dedicated Metrics page on the DB console displaying metrics related to Cross-Cluster Replication. 

Epic: CRDB-18752

Jira issue: CRDB-23888

### PR DESCRIPTION
**Pull request by adityamaru**
This change adds a new option to the metric dropdown dashboard
for Cross-Cluster Replication. This page will show all the user-facing
metrics relevant to C2C replication.

Informs: https://github.com/cockroachdb/cockroach/issues/95995

Release note: None

server,db-console: add feature flag for Cross-Cluster Replication dashboard

This change adds a feature flag whose value is determined by the cluster
setting `cross_cluster_replication.enabled`. This feature flag is then used
when rendering the metrics dashboard dropdown options to decide whether to
display the cross-cluster replication dashboard.

This change also deletes the session variable and associated cluster setting
`sql.defaults.experimental_stream_replication.enabled` so as to unify all
interactions with cross-cluster replication under a single cluster setting.

Fixes: https://github.com/cockroachdb/cockroach/issues/95995

Release note: None

### CODE REVIEW

**Comment by adityamaru** [comment]
Adding some screenshots of the above change:

- With `set cluster setting cross_cluster_replication.enabled = false;`

Screenshot 2023-02-01 at 6 27 37 PM

- With `set cluster setting cross_cluster_replication.enabled = true;`
Screenshot 2023-02-01 at 6 28 04 PM


Screenshot 2023-02-01 at 6 28 17 PM
Screenshot 2023-02-01 at 6 28 22 PM

**Comment by adityamaru** [comment]
I've been talking to @benbardin about this diff so I expect him to lead the review, but @stevendanna / @lidorcarmel it would be nice to get your LGTM on some of the C2C cluster setting + metrics changes. Thanks!

File: pkg/server/server_http.go [hunk 1]
```diff
@@ -17,8 +17,10 @@ import (
 
 	"github.com/NYTimes/gziphandler"
 	"github.com/cockroachdb/cmux"
+	"github.com/cockroachdb/cockroach/pkg/ccl/streamingccl"
```
Comment (adityamaru) [inline pkg/server/server_http.go:20]: oh this is no bueno, i'll have to inject this value or move the cluster setting to a non-ccl location.

**Review by dhartunian — COMMENTED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @benbardin, @lidorcarmel, @srosenberg, and @stevendanna)

___
*[`pkg/ccl/streamingccl/settings.go` line 19 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHt6tS7Zh1Irja-APH:-NNHt6tS7Zh1Irja-API:bt917a4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5cde7e2ec915dae50f6b995d63d453c8ff489d69/pkg/ccl/streamingccl/settings.go#L19)):*
> ```Go
> // CrossClusterReplicationEnabled enables the ability to setup and control a
> // cross cluster replication stream.
> var CrossClusterReplicationEnabled = settings.RegisterBoolSetting(
> ```

if you wanted to, you could also read the cluster setting in the frontend instead of also wiring up a feature flag. the cluster settings could be more easily refreshed at runtime in the client as well.

if you look in `clusterSettings.selectors.ts` there are examples of cluster setting redux selectors you could wire up into `nodeGraphs`.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 109 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrAGwEPs12QXsJB43:-NNHrAGx64vF_nQKGf1g:bn93hjz) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L109)):*
> ```XML
>   crossClusterReplication: {
>     label: "Cross-Cluster Replication",
>     component: crossClusterReplication,
> ```

nit: just for consistency, calling this a `*Dashboard` would be nice.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrqV_CRamW2P0Mi17:-NNHrqVa-iAWZ3w-5m8u:blpn6uh) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L39)):*
> ```XML
>     </LineGraph>,
>     <LineGraph
>       title="KV Bytes"
> ```

do the byte computations need to be measured as "rates"? they're monotonic counters, right?
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 48 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHs3TzBuPigVc9mHxi:-NNHs3TzBuPigVc9mHxj:bp8dsyk) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L48)):*
> ```XML
>           title="KV Bytes"
>           downsampleMax
>           aggregateMax
> ```

you might want `aggregateSum` here. The aggregator decides how metrics from multiple nodes are aggregated together when someone looks at a "cluster" view of the metrics. I think in the case of byte counters you want to add them. I believe sum is the default aggregator (see `queryFromProps` in `metricDataProvider/index.tsx`) so you can also omit it.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrH6H5etYoUUIHwgW:-NNHrH6H5etYoUUIHwgX:bl0lxq3) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L55)):*
> ```XML
>       title="SST Bytes"
>       sources={storeSources}
>       tooltip={`SST bytes (compressed) sent to KV by all replication jobs`}
> ```

do customers know how to disambiguate `KV bytes` and `SST bytes`?
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 62 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHsBjq7KkJnJsHcd3_:-NNHsBjq7KkJnJsHcd3a:b-bzot9c) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L62)):*
> ```XML
>           title="SST Bytes"
>           downsampleMax
>           aggregateMax
> ```

same as above, probably want "sum".

**Review by benbardin — COMMENTED** [review]
nothing to add beyond david's comments and your own! though I'd certainly appreciate somebody who knows streamingest better reading that directory.

**Review by adityamaru — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian, @lidorcarmel, @srosenberg, and @stevendanna)

___
*[`pkg/ccl/streamingccl/settings.go` line 19 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHt6tS7Zh1Irja-APH:-NNNA_LXBxv4I9JtSDFa:b-ihy99d) ([raw file](https://github.com/cockroachdb/cockroach/blob/5cde7e2ec915dae50f6b995d63d453c8ff489d69/pkg/ccl/streamingccl/settings.go#L19)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

if you wanted to, you could also read the cluster setting in the frontend instead of also wiring up a feature flag. the cluster settings could be more easily refreshed at runtime in the client as well.

if you look in `clusterSettings.selectors.ts` there are examples of cluster setting redux selectors you could wire up into `nodeGraphs`.
</blockquote></details>

Thanks for this, it makes the change smaller! I did have to move the filter call into `render` instead of leaving it in the constructor to get things working. I did this because we refresh the settings in `componentDidMount` and  so the state is only set after that method is invoked. I'm very new to redux/react so let me know if there is a better place I should move the filter so that it is not invoked on each render.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 109 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrAGwEPs12QXsJB43:-NNMeiVr4l8VbBlvsiy1:b-v20wi7) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L109)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

nit: just for consistency, calling this a `*Dashboard` would be nice.
</blockquote></details>

ahh yes, fixed.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrqV_CRamW2P0Mi17:-NNMg1n74tUCkyu_G1m7:bj2w4ms) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L39)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

do the byte computations need to be measured as "rates"? they're monotonic counters, right?
</blockquote></details>

They are monotonic counters. After reading `Snapshot Data Received` in `replication.tsx` I realize I should add `nonNegativeRate` right? 
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 48 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHs3TzBuPigVc9mHxi:-NNMfCdv5MqnEW-YNEdB:b-xowie6) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L48)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

you might want `aggregateSum` here. The aggregator decides how metrics from multiple nodes are aggregated together when someone looks at a "cluster" view of the metrics. I think in the case of byte counters you want to add them. I believe sum is the default aggregator (see `queryFromProps` in `metricDataProvider/index.tsx`) so you can also omit it.
</blockquote></details>

Yup you're right, thanks. I believe I can omit both the downsample and aggregate params.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrH6H5etYoUUIHwgW:-NNMgJjLDq6o6fwJi6Qj:bixfpc5) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L55)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

do customers know how to disambiguate `KV bytes` and `SST bytes`?
</blockquote></details>

Good question, and I think the answer is no. Another alternative is to refer to them as logical bytes and physical bytes. We do have precedence for these terms in https://www.cockroachlabs.com/docs/stable/ui-storage-dashboard.html, @lidorcarmel curious what you think?
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 62 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHsBjq7KkJnJsHcd3_:-NNMfX2A0A2ywczCDORk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L62)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

same as above, probably want "sum".
</blockquote></details>

Done.

**Review by dhartunian — COMMENTED** [review]
> Reviewed 20 of 20 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @lidorcarmel, @srosenberg, and @stevendanna)

___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 176 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNbV3aKBwIuDWB8sn4k:-NNbV3aKBwIuDWB8sn4l:b-xgm78t) ([raw file](https://github.com/cockroachdb/cockroach/blob/14a43b5faa3352ff915c1d38a39d8604f5321975/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L176)):*
> ```XML
>   refresh = () => {
>     this.props.refreshNodes();
>     this.props.refreshLiveness();
> ```

consider adding a `refreshSettings()` prop here (via the Selectors), and calling it. This will ensure that settings changes are refreshed when the metrics page is loaded since we now depend on them. will hopefully prevent one or two support tickets where ppl don't see what they expect.

This prop will be added in the `mapDispatchToProps` redux helper. (MapStateToProps lets your component access data from the Redux store, and MapDispatchToProps gives your component handles to action dispatchers that can mutate the redux store)
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 332 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNbSjge8xmrQ3VsCxku:-NNbSjge8xmrQ3VsCxkv:b-m5987o) ([raw file](https://github.com/cockroachdb/cockroach/blob/14a43b5faa3352ff915c1d38a39d8604f5321975/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L332)):*
> ```XML
>       : dashboardDropdownOptions.filter(
>           option => option.label !== "Cross-Cluster Replication",
>         );
> ```

This is fine to put here. If the list was long, or the filter was complex, it would be worth memoizing but there's plenty of other stuff slowing the code down here already to load and render metrics.

What you did is actually recommended here https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization until you hit performance constraints.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrqV_CRamW2P0Mi17:-NNbRQ-J1eagwJkvKjeB:b4c9lyr) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L39)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

They are monotonic counters. After reading `Snapshot Data Received` in `replication.tsx` I realize I should add `nonNegativeRate` right? 
</blockquote></details>

:thumbsup: 
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrH6H5etYoUUIHwgW:-NNbRacZ8mhi3Wpe3JFF:b4dmnh8) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L55)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

Good question, and I think the answer is no. Another alternative is to refer to them as logical bytes and physical bytes. We do have precedence for these terms in https://www.cockroachlabs.com/docs/stable/ui-storage-dashboard.html, @lidorcarmel curious what you think?
</blockquote></details>

SGTM. Especially since we have documentation around the difference.

**Review by lidorcarmel — APPROVED** [review]
streamingest stuff lgtm.

**Review by lidorcarmel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @dhartunian, @srosenberg, and @stevendanna)

___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrH6H5etYoUUIHwgW:-NNbmZoU0ZH8rw8hW0Uv:b-y2ntvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L55)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

SGTM. Especially since we have documentation around the difference.
</blockquote></details>

the kv bytes can be renamed to logical, I think it's accurate.

but the sst bytes is not exactly physical - it is the number of bytes in ssts that replication sends to kv, which is not really the physical bytes: physical is (I think) the bytes written to the disk. the sst bytes are different because kv may uncompress the sst (if it's small) and also, pebble may compact which will write more to the disk. I think it's ok to have the name say exactly what we do in this case.

**Review by lidorcarmel — COMMENTED** [review]
@adityamaru you might want to update some text in `chart_catalog.go` for those metric renames.

**Review by adityamaru — COMMENTED** [review]
Yup nice catch, done!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-:-NNd2ikZ0dEouAVnFS0m:bdtxopu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian, @lidorcarmel, @srosenberg, and @stevendanna)

___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 176 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNbV3aKBwIuDWB8sn4k:-NNd19uFExAg9UZaiTXY:b-yw2jkb) ([raw file](https://github.com/cockroachdb/cockroach/blob/14a43b5faa3352ff915c1d38a39d8604f5321975/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L176)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

consider adding a `refreshSettings()` prop here (via the Selectors), and calling it. This will ensure that settings changes are refreshed when the metrics page is loaded since we now depend on them. will hopefully prevent one or two support tickets where ppl don't see what they expect.

This prop will be added in the `mapDispatchToProps` redux helper. (MapStateToProps lets your component access data from the Redux store, and MapDispatchToProps gives your component handles to action dispatchers that can mutate the redux store)
</blockquote></details>

We already have a `refreshNodeSettings` property. Following it through it seems to be driven by the `refreshSettings` in `apiReducers.ts` which looks like what we want.
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 332 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNbSjge8xmrQ3VsCxku:-NNd1XW16FBWOL36MYkI:b-iyss4b) ([raw file](https://github.com/cockroachdb/cockroach/blob/14a43b5faa3352ff915c1d38a39d8604f5321975/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L332)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

This is fine to put here. If the list was long, or the filter was complex, it would be worth memoizing but there's plenty of other stuff slowing the code down here already to load and render metrics.

What you did is actually recommended here https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization until you hit performance constraints.
</blockquote></details>

perfect, thanks
___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx` line 55 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNHrH6H5etYoUUIHwgW:-NNd2em06tg6By4Trbzd:bw694n4) ([raw file](https://github.com/cockroachdb/cockroach/blob/79f6e5bd04e073f8d2099251f41b2eb53195d57f/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/crossClusterReplication.tsx#L55)):*
<details><summary><i>Previously, lidorcarmel (Lidor Carmel) wrote…</i></summary><blockquote>

the kv bytes can be renamed to logical, I think it's accurate.

but the sst bytes is not exactly physical - it is the number of bytes in ssts that replication sends to kv, which is not really the physical bytes: physical is (I think) the bytes written to the disk. the sst bytes are different because kv may uncompress the sst (if it's small) and also, pebble may compact which will write more to the disk. I think it's ok to have the name say exactly what we do in this case.
</blockquote></details>

makes sense to me, changed kv bytes to logical bytes.

**Review by dhartunian — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-:-NNgLOLW8WQYqV_a57rB:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @adityamaru, @lidorcarmel, @srosenberg, and @stevendanna)

___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx` line 176 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/96206#-NNbV3aKBwIuDWB8sn4k:-NNgLBJjAsJRjbz1ehE_:b-hg8a2n) ([raw file](https://github.com/cockroachdb/cockroach/blob/14a43b5faa3352ff915c1d38a39d8604f5321975/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/index.tsx#L176)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

We already have a `refreshNodeSettings` property. Following it through it seems to be driven by the `refreshSettings` in `apiReducers.ts` which looks like what we want.
</blockquote></details>

ah nice! I missed that. :thumbsup:

**Comment by adityamaru** [comment]
TFTRs!

bors r+

**Comment by rafiss** [comment]
@adityamaru exciting change! since this is adding a user-visible page to the DB console, it needs a release note. or since this was already merged, we'd need to make a DOCS ticket so that the page gets documented at https://www.cockroachlabs.com/docs/v22.2/ui-overview

**Comment by adityamaru** [comment]
>it needs a release note. or since this was already merged, we'd need to make a DOCS ticket so that the page gets documented

Since we only plan on releasing C2C for private preview in 23.1 we've decided not to have public-facing documentation or release notes for any of the work around this new feature. In the future when we do decide to release it publically we'll have to ensure we document this along with many of the other changes that have gone in over the past release.

**Pull request merged**